database/rrdvar.c

```diff
@@ -227,17 +241,19 @@ int health_variable_lookup(const char *variable, uint32_t hash, RRDCALC *rc, cal
         return 1;
     }
 
-    rv = rrdvar_index_find(&st->rrdfamily->rrdvar_root_index, variable, hash);
-    if(rv) {
-        *result = rrdvar2number(rv);
+    struct rrdcalc_rrdset_alarm search;
+    search.st = st;
+    int ret = health_variable_lookup_test(result, &search, variable, hash, &host->alarms_idx_health_name);
+    if(ret)
         return 1;
-    }
 
-    rv = rrdvar_index_find(&host->rrdvar_root_index, variable, hash);
-    if(rv) {
-        *result = rrdvar2number(rv);
+    ret = health_variable_lookup_test(result, &search, variable, hash, &host->alarms_idx_health_family);
+    if(ret)
+        return 1;
+
+    ret = health_variable_lookup_test(result, &search, variable, hash, &host->alarms_idx_health_hostid);
+    if(ret)
```

Comment:
What about all the variables that are in `st->rrdfamily->rrdvar_root_index` and `host->rrdvar_root_index`? Are they lost forever? For example, dimension variables, see here:
https://github.com/netdata/netdata/blob/cb7f19ba2dc40d24ec9cd68c08a7d20acd891a37/database/rrddimvar.c#L134-L151

Reply:
Of course no, you are correct, I fix the problem now.

Thank you!